Source/WebCore/rendering/RenderBlockFlow.h

```diff
@@ -181,6 +181,12 @@ class RenderBlockFlow : public RenderBlock {
         bool m_hasMarginAfterQuirk : 1;
         bool m_determinedMarginBeforeQuirk : 1;
 
+        // Used to cache the value of canTrimBlockStartMargins inside RenderLayoutState. Nested blocks
+        // could set the value in RenderLayoutState to false in order to make sure no more trimming occurs
+        // beyond that point, but we still need to make sure the margins of our children at the block-start
+        // edge have their margins trimmed.
```

Comment:
Could you explain (maybe with a test case) at what point would the layout state value be different from this margin info value? It looks like you track 2 states and they may or may not have the same value and I am just curious where in block layout/margin collapsing would they diverge.

Reply:
Sure, here is a small test case:

```
<div id="container" style="margin-trim: block">
    <div id="outer" style="margin-block-start: 10px; border: 1px solid black; width: 50px; height: 50px; background-color: green;">
        <div id="inner" style="margin-block-start: 10px; border: 1px solid black; width: 50px; height: 50px; background-color: blue;"></div>
    </div>
</div>
```
The outer div "container" will set the trimming layout state to true to indicate that margins at the block-start should be trimmed. Before it recurses into the layout for "outer" it will cache that state inside `MarginInfo`. "outer" will see that its block-start margins can't collapse through because of the trimming so it sets the value in layout state value to false. By the time we get back into container its cached value differs from the value in layout state because of this